homeassistant/components/asuswrt/config_flow.py

```diff
@@ -101,71 +130,97 @@ def _get_ip(host: str) -> str | None:
 
 
 class AsusWrtFlowHandler(ConfigFlow, domain=DOMAIN):
-    """Handle a config flow."""
+    """Handle a config flow for AsusWRT."""
 
     VERSION = 1
 
+    def __init__(self) -> None:
+        """Initialize the AsusWrt config flow."""
+        self._config_data: dict[str, Any] = {}
+
     @callback
-    def _show_setup_form(
-        self,
-        user_input: dict[str, Any] | None = None,
-        errors: dict[str, str] | None = None,
-    ) -> FlowResult:
+    def _show_setup_form(self, error: str | None = None) -> FlowResult:
         """Show the setup form to the user."""
 
-        if user_input is None:
-            user_input = {}
+        user_input = self._config_data
 
-        adv_schema = {}
-        conf_password = vol.Required(CONF_PASSWORD)
         if self.show_advanced_options:
-            conf_password = vol.Optional(CONF_PASSWORD)
-            adv_schema[vol.Optional(CONF_PORT)] = cv.port
-            adv_schema[vol.Optional(CONF_SSH_KEY)] = str
+            add_schema = {
+                vol.Exclusive(CONF_PASSWORD, PASS_KEY, PASS_KEY_MSG): str,
+                vol.Optional(CONF_PORT): cv.port,
+                vol.Exclusive(CONF_SSH_KEY, PASS_KEY, PASS_KEY_MSG): str,
+            }
+        else:
+            add_schema = {vol.Required(CONF_PASSWORD): str}
 
         schema = {
             vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
             vol.Required(CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")): str,
-            conf_password: str,
-            vol.Required(CONF_PROTOCOL, default=PROTOCOL_SSH): vol.In(
-                {PROTOCOL_SSH: "SSH", PROTOCOL_TELNET: "Telnet"}
-            ),
-            **adv_schema,
-            vol.Required(CONF_MODE, default=MODE_ROUTER): vol.In(
-                {MODE_ROUTER: "Router", MODE_AP: "Access Point"}
+            **add_schema,
+            vol.Required(
+                CONF_PROTOCOL,
+                default=user_input.get(CONF_PROTOCOL, PROTOCOL_HTTPS),
```

Comment:
If HTTPS is unavailable, will the integration raise an error or automatically fallback to HTTP?

Reply:
No fallback to HTTP (it should be a security issue, right?) If HTTPS is not available the connection fail and the integration raise an error.

Reply:
Perfect :) We should indeed **not** fallback to http